src/Symfony/Component/String/AbstractString.php

```diff
@@ -620,14 +620,20 @@ public function toUnicodeString(): UnicodeString
     /**
      * @return static
      */
-    public function truncate(int $length, string $ellipsis = ''): self
+    public function truncate(int $length, string $ellipsis = '', bool $preserveWord = false): self
     {
         $stringLength = $this->length();
 
         if ($stringLength <= $length) {
             return clone $this;
         }
 
+        if ($preserveWord) {
+            $spacePosition = $this->indexOf(' ', $length);
```

Comment:
The main issue I can see is the definition of a "word". Some languages do not use spaces between "words".

Reply:
Interesting, I'll try to take a look at how the `cut` parameter works in `wordwrap`.

Reply:
I've looked around in other programming languages and libraries and all of them used the normal space separator (and some, the `\s` regexp separator). Even PHP uses the simple space in functions like `wordwrap()` (see https://github.com/php/php-src/blob/4c9699f0d1890ea5ba3d821783f335637f35d209/ext/standard/string.c#L952).

We could do the same ... although important languages like Japanese and Chinese don't use spaces to separate words (see https://en.wikipedia.org/wiki/Line_breaking_rules_in_East_Asian_languages).